src/client.lisp

```diff
@@ -0,0 +1,95 @@
+;;;; client.lisp
+;;;;
+;;;; Author: Eric Peterson
+;;;;
+;;;; Lisp mimic of the python JSON RPC Client.
+
+
+(in-package #:rpcq)
+
+(defstruct rpc-client
+  "Holds the data for an (active) RPCQ client connection."
+  socket
+  timeout)
+
+(define-condition RPC-error (simple-error)
+  ((error-string :initarg :string :reader RPC-error-string)
+   (request-id :initarg :id :reader RPC-error-request-id))
+  (:report (lambda (condition stream)
+             (format stream "RPC request ~a resulted in error:~%~a~&"
+                     (RPC-error-request-id condition)
+                     (RPC-error-string condition)))))
+
+(define-condition RPC-protocol-error (simple-error)
+  ((object :initarg :object :reader RPC-protocol-error-object)
+   (id :initarg :id :reader RPC-protocol-error-id))
+  (:report (lambda (condition stream)
+             (format stream "RPC request ~a resulted in bad reply:~%~a~&"
+                     (RPC-protocol-error-id condition)
+                     (RPC-protocol-error-object condition)))))
+
+
+(defun rpc-call (client call &rest args)
+  "Makes a synchronous RPC client call over a client connection.  Returns the result of the call directly."
+  (let* ((uuid (unicly:uuid-princ-to-string (unicly:make-v4-uuid)))
+         (request (make-instance '|RPCRequest|
+                                 :|id| uuid
+                                 :|params| (alexandria:plist-hash-table args :test #'equal)
+                                 :|method| (sanitize-name call)))
+         (payload (serialize request))
+         (socket (rpc-client-socket client)))
+    (cffi:with-foreign-object (foreign-payload :unsigned-char (length payload))
```

Comment:
Why `unsigned-char`? Is that demanded by `pzmq:send`?

Also, while perhaps not for this PR, we could optimize all of this by eliminating the copying of `payload` to foreign memory, and just get a pointer to the data directly if we can make guarantees about `serialize`.

Reply:
If it's a vector of integers 0 <= x < 256 or <= 256?